src/com/sap/piper/Utils.groovy

```diff
@@ -163,5 +163,12 @@ static String evaluateFromMavenPom(Script script, String pomFileName, String pom
         defines: "-Dexpression=$pomPathExpression -DforceStdout -q",
         returnStdout: true
     )
+    if (resolvedExpression.startsWith('null object or invalid expression')) {
+        // There is no error indication (exit code or otherwise) from the
+        // 'evaluate' Maven plugin, only this output to stdout. The calling
+        // code assumes an empty string is returned when the property could
+        // not be resolved.
+        resolvedExpression = ''
```

Comment:
Wouldn't it be better to raise an exception in case we are not able to retrieve the version (e.g. `throw new HudsonAbortException('Cannot retrieve version from \"${pomFileName}\"")?

IMO having the empty string as version is only slight better compared to having the `null object or invalid expression` as version string. But it doesn't look like an appropriate approach.

Are we able to continue in a reasonable way after being unable to retrieve the version (e.g. is there a fallback - I can't see any)? When we bubble up the call stack we end up in step `artifactSetVersion`. I can't see any check for empty string there so I guess we would continue with setting an empty version. In the best case the build fails somewhere later on (violating the fail-fast principle). In the bad case we release some artifacts with an empty version.

Reply:
Thanks for the review! I agree in general and will have a look at the calling code in the SDK pipeline to see what error handling is implemented there.

Reply:
I am also in favor of throwing an exception

Reply:
Done